rosbag2_cpp/include/rosbag2_cpp/readers/sequential_reader.hpp

```diff
@@ -130,10 +132,13 @@ class ROSBAG2_CPP_PUBLIC SequentialReader
   std::unique_ptr<Converter> converter_{};
   std::unique_ptr<rosbag2_storage::MetadataIo> metadata_io_{};
   rosbag2_storage::BagMetadata metadata_{};
+  std::vector<rosbag2_storage::TopicMetadata> topics_metadata_{};
   std::vector<std::string> file_paths_{};  // List of database files.
   std::vector<std::string>::iterator current_file_iterator_{};  // Index of file to read from
 
 private:
+  void fill_topics_and_types();
```

Comment:
does this have to be part of a public API?

Reply:
Under this philosophy, are private methods ever appropriate?

Reply:
haha, maybe I read too much Scott Meyers' effective c++. 
I am not totally against private functions. I just believe in free functions if easy possible. In this case, this function is getting called once per application and doesn't really depend on an internal runtime state of the class, except the bag metadata which could easily be passed in as an argument.